src/avlizer_confluent.erl

```diff
@@ -125,11 +131,22 @@ stop() ->
 %% number (or a stream) of objects having the same reference.
 -spec make_decoder(ref()) -> avro:simple_decoder().
 make_decoder(Ref) ->
-  Schema = maybe_download(Ref),
-  avro:make_simple_decoder(Schema, []).
+  do_make_decoder(Ref, ?DEFAULT_CODEC_OPTIONS).
+
+-spec make_decoder(name(), fp() | codec_options()) -> avro:simple_decoder().
+make_decoder(Ref, CodecOptions) when is_list(CodecOptions) ->
+  do_make_decoder(Ref, CodecOptions);
+make_decoder(Name, Fp) ->
+  do_make_decoder({Name, Fp}, ?DEFAULT_CODEC_OPTIONS).
```

Comment:
Unfortunately I think the specs are wrong. The code very explicitly allows and handles fingerprints as plain lists (see `unify_ref/1` and `fp_to_subject/2`) and I think we need to preserve that. So when the 2nd parameter here is a list, you need to perform a test to see if it's a fingerprint or not, for example:
```
make_decoder(Ref, FpOrCodecOptions) when is_list(FpOrCodecOptions) ->
  case is_fp(FpOrCodecOptions) of
    true -> ... ;
    false -> ...
  end;
...
is_fp(List) ->
  try
    _ = iolist_to_binary(fp),
    true
  catch
    error:badarg ->
      false
  end.
```
and similar in the other API entry points.

Alternatively only handle CodecOptions in the new `/3` entry points, but allow passing in e.g. `false` as Fp and in that case don't include that in the Ref.

Maybe @zmstone has some input, as the author of this?